### PR TITLE
Handles signaling process run as sudo

### PR DESCRIPTION
**Pull request by cacarrara**
This change makes possible to call `kill_process_tree()` function with
pid of "sudo processes" and it also makes possible handle timeout of
subprocesses called with `sudo=True`.

Both cases now use a new function created in process module to return
the user owner of a pid (`get_user_of_pid()`).

For the case of subprocesses run with sudo=True it was necessary adjust
the `send_signal()` method of `SubProcess` class so it can send the signal
for sudo process child.

Reference: https://trello.com/c/ROPWdFFr
Signed-off-by: Caio Carrara <ccarrara@redhat.com>

### CODE REVIEW

**Comment by cacarrara** [comment]
It was taking some time to figure it out the best approach for it and how to test it as well. So I decided to open the PR to get some feedbacks and/or if you think it's already good enough, we can merge it.

File: avocado/utils/process.py [hunk 1]
```diff
@@ -1553,3 +1579,16 @@ def getstatusoutput(cmd, timeout=None, verbose=False, ignore_status=True,
     if text[-1:] == '\n':
         text = text[:-1]
     return (sts, text)
+
+
+def get_user_of_pid(pid):
```
Comment (clebergnu) [inline avocado/utils/process.py]: This function seems useful, and can certainly be introduced in its own commit.

File: avocado/utils/process.py [hunk 2]
```diff
@@ -791,6 +803,20 @@ def get_pid(self):
...
+        """
+        self._init_subprocess()
+        return get_user_of_pid(self.get_pid())
+
+    def is_running_as_sudo(self):
+        """
+        Returns whether a process is running as sudo
+        """
+        self._init_subprocess()
+        return self.get_user() == 'root'
```
Comment (clebergnu) [inline avocado/utils/process.py]: Even though this *usually* works, it's not be the condition for an UNIX process to be considered privileged.  The process effective user id being 0 is the definitive check.  What I mean is, you can change the name of your `root` user, but as long as it has uid == 0, it will be able to perform privileged actions.

File: avocado/utils/process.py [hunk 3]
```diff
@@ -750,7 +756,13 @@ def send_signal(self, sig):
         :param sig: Signal to send.
         """
         self._init_subprocess()
-        self._popen.send_signal(sig)
+        if self.is_running_as_sudo():
+            for child in get_children_pids(self.get_pid()):
+                child_pid = child.decode('UTF-8')
```
Comment (clebergnu) [inline avocado/utils/process.py]: `get_children_pid` is supposed to return a `list of int`.  If it is, there's no need for this line here.  If it's not, it needs to be fixed.

File: avocado/utils/process.py [hunk 4]
```diff
@@ -750,7 +756,13 @@ def send_signal(self, sig):
...
+                kill_child_cmd = 'kill -%d %s' % (sig.value, child_pid)
+                run(kill_child_cmd, sudo=True)
```
Comment (clebergnu) [inline avocado/utils/process.py]: Maybe we need to treat (ignore some) errors here because there may be race conditions where some of the processes that we got on `get_children_pid()` may not exist anymore.

**Comment by cacarrara** [comment]
This PR version definetly is not going to be merged so I'm already closing it. I'm going to rework on the reviewed points and continue purposing in a new version. Thanks.
